administrator/components/com_content/controllers/article.php

```diff
@@ -145,6 +146,30 @@ public function batch($model = null)
 	}
 
 	/**
+	 * Method to generate and store share token.
+	 *
+	 * @return  boolean   True if token successfully stored, false otherwise and internal error is set.
+	 *
+	 * @since   3.7
+	 */
+	public function shareDraft()
+	{
+		if (!JSession::checkToken('get'))
+		{
+			$this->app->enqueueMessage(JText::_('JINVALID_TOKEN'));
+			$this->app->redirect('index.php');
+		}
+
+		$title = $this->input->getArray(array('jform' => array('title' => 'string')));
+
+		// Get the model
+		$model = $this->getModel();
+		$return = $model->shareToken($title);
+		echo new JResponseJson($return);
```

Comment:
for eg like this ?
echo new JResponseJson($return, JText::_('MY_TASK_ERROR'), true); 
the last parameter is to manually set the error flag, should the last parameter be there too?